lib/torrent.js

```diff
@@ -207,6 +207,12 @@ Torrent.prototype._onParsedTorrent = function (parsedTorrent) {
     })
   }
 
+  // destroy DHT client if the torrent is private
+  if (self.parsedTorrent.private && self.client.dht) {
+    self.client.dht.destroy()
+    self.client.dht = null
+  }
```

Comment:
Other torrents might not be private, and we want the DHT to be available to them. Best to solve this a different way that preserves the DHT for use by other torrents.